internal/agent/hooks/gruboptions.go

```diff
@@ -31,10 +33,32 @@ func (b GrubPostInstallOptions) Run(c config.Config, _ v1.Spec) error {
 		return nil
 	}
 	c.Logger.Logger.Debug().Msg("Running GrubOptions hook")
-	err := system.Apply(system.SetGRUBOptions(c.GrubOptions))
+	c.Logger.Debugf("Setting grub options: %s", c.GrubOptions)
+	err := grubOptions(c, c.GrubOptions)
 	if err != nil {
-		c.Logger.Logger.Error().Err(err).Msg("Failed to set grub options")
+		return err
 	}
-	c.Logger.Logger.Debug().Msg("Running GrubOptions hook")
+	c.Logger.Logger.Debug().Msg("Finish GrubOptions hook")
 	return nil
 }
+
+// grubOptions sets the grub options in the grubenv file
+// It mounts the OEM partition if not already mounted
+// If its mounted but RO, it remounts it as RW
```

Comment:
I don't see this anywhere in the code below (checking if it's mounted RO)

Reply:
`mounts.PrepareWrite` will do that for us, if its mounted and in the same mountpoint it will do nothing, if not mounted it will mount it and if mounted but RO it will remount it RW.

Reply:
but you only call `mounts.PrepareWrite` inside `if !runtime.OEM.Mounted {`. So if it's mounted, `PrepareWrite` is not called at all?

Reply:
yes, because we by default mount oem as RW always, if its already mounted we assume its already RW. maybe we should change that?

I dont know, I dont like that this hook has this special mount thing while the rest have a strigth syscall.mount IIRC so Ill change this anyway to fall in line wiht others. ITs just that its what it was used already so I did not bother going deeper :D